src/LibraryManager/Providers/FileSystem/FileSystemProvider.cs

```diff
@@ -100,14 +110,56 @@ public ILibraryCatalog GetCatalog()
         }
 
         /// <summary>
-        /// No-op for FileSystemProvider
+        /// Updates file set on the passed in ILibraryInstallationState in case user selected to have all files included
         /// </summary>
         /// <param name="desiredState"></param>
         /// <param name="cancellationToken"></param>
         /// <returns></returns>
-        public Task<ILibraryInstallationResult> UpdateStateAsync(ILibraryInstallationState desiredState, CancellationToken cancellationToken)
+        public async Task<ILibraryInstallationResult> UpdateStateAsync(ILibraryInstallationState desiredState, CancellationToken cancellationToken)
         {
-            return Task.FromResult<ILibraryInstallationResult>(LibraryInstallationResult.FromSuccess(desiredState));
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return LibraryInstallationResult.FromCancelled(desiredState);
+            }
+
+            try
+            {
+                var catalog = (FileSystemCatalog)GetCatalog();
```

Comment:
Can we use "as" instead of casting to avoid exception here and check if catalog is null?

Reply:
Actually GetCatalog() returns a new instance of FileSystemCatalog so it is ok.

Reply:
Do we really need to cast it here? GetCatalog() returns ILibraryCatalog, which has GetLibraryAsync() on it already. 

In general though, I agree with Bhavya here. Even if it's not needed in the current implementation, it's future-proof and cleaner. 

Anyway - the right change here seems to be to remove the cast and simply use ILibraryCatalog as the variable type (and not "var" :) ).